### ISSUE TITLE
Aiojobs stops processing jobs when CancelledError occurs.

### ISSUE THREAD

**Issue by F1ashhimself** [aio-libs/aiojobs#97]
aiohttp==3.5.4
aiojobs==0.2.2

Server example:
```python
import asyncio

from aiohttp import web
from aiojobs.aiohttp import setup, spawn


async def bar():
    await asyncio.sleep(1)


async def foo(request):
    await spawn(request, bar())
    return web.Response(text="Foo")


@web.middleware
async def error_middleware(request, handler):
    try:
        return await handler(request)
    except asyncio.CancelledError:
        print('CancelledError occurred.')
        raise

app = web.Application()
app.router.add_get('/foo', foo, name='index')
app.middlewares.append(error_middleware)
setup(app, pending_limit=100, limit=10)

web.run_app(app)
```

Then Im running `ab -c 100 -n 200 "http://127.0.0.1:8080/foo"` to load application.
So when the `pending_limit` "pool" will be full we will receive `CancelledError` and then we wont be able to spawn new jobs. Previously spawned jobs will be not processed too so `http://127.0.0.1:7000/foo` won't work till application restart. Also aiojobs scheduler is not closed at this moment.

**Comment by kolosovi** [comment]
Is this issue related to CancelledError? I was able to make aiojobs scheduler hang without CancelledError being raised. See #98 , probably it's the same issue as yours.
Scheduler will hang when too many jobs arrive within a short time span and your description resembles this case.

**Comment by pollydrag** [comment]
Fixed by #135
